paddle/gserver/layers/ContextProjection.h

```diff
@@ -41,7 +41,7 @@ namespace paddle {
  *
  * The config file api is context_projection.
  */
-class ContextProjection : public Projection {
+class ContextProjection : public Projection, public WithFunction {
```

Comment:
Done. I use `FunctionList` instead of `WithFunction`, make `FunctionList` as a member of Layer and Projection.